internal/upload/azure/azure.go

```diff
@@ -34,6 +35,11 @@ type ImageMetadata struct {
 // It can speed up the upload by using goroutines. The number of parallel goroutines is bounded by
 // the `threads` argument.
 func UploadImage(credentials Credentials, metadata ImageMetadata, fileName string, threads int) error {
+	// Azure cannot create an image from a storage blob without .vhd extension
```

Comment:
Is the requirement really on the name of the storage blob, or the format? Not a blocker either way, but I think down the line we should make a few changes:
 - make the upload target per image-type in our distro package, so that it is not possible to upload to the wrong one, if that makes sense.
 - if the requirement is about the format, rather than the name, then drop this check again when the format is "correct by construction".

Reply:
The storage blob must be in the `vhd` format and also its name must end with the `.vhd` extension (otherwise the UI complains).

It makes sense to have the upload target per an image type, we should definitely do it at some point (I think it really only makes sense to upload AMI image type to AWS and VHD to Azure).

Reply:
Thanks! Agreed.